gpiozero/output_devices.py

```diff
@@ -1264,16 +1284,20 @@ def backward(self, speed=1):
 
         :param float speed:
             The speed at which the motor should turn. Can be any value between
-            0 (stopped) and the default 1 (maximum speed) if *pwm* was
-            :data:`True` when the class was constructed (and only 0 or 1 if
-            not).
+            0 (stopped) and the default 1 (maximum speed) if *pwm* or *enable_is_pwm*
+            was :data:`True` when the class was constructed (and only 0 or 1 if not).
         """
         if not 0 <= speed <= 1:
             raise ValueError('backward speed must be between 0 and 1')
-        if isinstance(self.backward_device, DigitalOutputDevice):
+        if isinstance(self.backward_device, DigitalOutputDevice) and isinstance(self.enable_device, DigitalOutputDevice):
             if speed not in (0, 1):
                 raise ValueError(
                     'backward speed must be 0 or 1 with non-PWM Motors')
+        if isinstance(self.enable_device, PWMOutputDevice):
+            self.forward_device.off()
+            self.enable_device.value = speed
+            self.backward_device.off()
```

Comment:
Should this be `self.backward_device.on()` ?? (and if so, how come your unit-tests didn't pick this up?)